apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl

```diff
@@ -474,3 +513,122 @@ maybe_install_wolff_telemetry_handlers(ResourceID) ->
         %% multiplying the metric counts...
         #{bridge_id => ResourceID}
     ).
+
+preproc_kafka_headers(HeadersTmpl) when HeadersTmpl =:= <<>>; HeadersTmpl =:= undefined ->
+    undefined;
+preproc_kafka_headers(HeadersTmpl) ->
+    case emqx_placeholder:preproc_tmpl(HeadersTmpl) of
+        [{var, _}] = Tks ->
+            Tks;
+        _ ->
+            ErrMsg = "The 'kafka_headers' must be a single placeholder like ${pub_props}",
```

Comment:
I think this would be best placed in a config validator, since it doesn't depend on runtime values.